### PR TITLE
Collision, position and math utilities protoype 

### PR DESCRIPTION
**Pull request by jakob753951**
Merge the prototype so the other teams can get started sooner

### CODE REVIEW

**Review by MGertz — APPROVED** [review]

File: Collision/src/main/java/dk/sdu/sesem4/collision/Collision.java
Comment (JfriisJ) [inline Collision/src/main/java/dk/sdu/sesem4/collision/Collision.java:1]: missing javadoc.

File: Collision/src/main/java/dk/sdu/sesem4/collision/CollisionSystem.java
Comment (JfriisJ) [inline Collision/src/main/java/dk/sdu/sesem4/collision/CollisionSystem.java:1]: missing javadoc.

File: Common/src/main/java/dk/sdu/sesem4/common/data/EntityParts/PositionPart.java
Comment (JfriisJ) [inline Common/src/main/java/dk/sdu/sesem4/common/data/EntityParts/PositionPart.java:1]: missing javadoc.

File: Common/src/main/java/dk/sdu/sesem4/common/data/math/Vector2.java [hunk 1]
```diff
@@ -5,37 +5,29 @@
...
+     * Creates a new Vector with the given x and y values
      */
     public Vector2(float x, float y) {
         this.x = x;
         this.y = y;
     }
 
     /**
-     * Creates a new vector with the given direction
+     * Creates a new Vector with magnitude 1 and the given direction
```
Comment (JfriisJ) [inline Common/src/main/java/dk/sdu/sesem4/common/data/math/Vector2.java:30]: missing param in javadoc

File: Common/src/test/java/dk/sdu/sesem4/common/data/EntityParts/PositionPartTest.java
Comment (JfriisJ) [inline Common/src/test/java/dk/sdu/sesem4/common/data/EntityParts/PositionPartTest.java:1]: missing javadoc.

File: Common/src/test/java/dk/sdu/sesem4/common/data/math/RectangleTest.java
Comment (JfriisJ) [inline Common/src/test/java/dk/sdu/sesem4/common/data/math/RectangleTest.java:1]: missing descriptions.

File: Map/src/main/java/dk/sdu/sesem4/map/MapProcessingService.java [hunk 2]
```diff
@@ -132,4 +140,31 @@ public void processNotification(Class<? extends EventType> eventType, Event data
 
 		eventData.getGameData().getGameWorld().setMap(getCurrentMap());
 	}
+
```
Comment (JfriisJ) [inline Map/src/main/java/dk/sdu/sesem4/map/MapProcessingService.java:143]: description missing.

**Review by JfriisJ — APPROVED** [review]
Remember the javadoc but looks fine with the changes

File: Common/src/test/java/dk/sdu/sesem4/common/data/math/RectangleTest.java [hunk 3]
```diff
@@ -0,0 +1,91 @@
+package dk.sdu.sesem4.common.data.math;
+import junit.framework.TestCase;
+
+
+public class RectangleTest extends TestCase {
```
Comment (The0mikkel) [inline Common/src/test/java/dk/sdu/sesem4/common/data/math/RectangleTest.java:5]: TestCase is JUnit 3

**Review by The0mikkel — CHANGES_REQUESTED** [review]
Some minor changes. Mostly documentation and testing.

I have heard somethgin about everything not being tested?

In the pom.xml files, JUnit 4 is used, but `extends TestCase` is a JUnit 3 way of declaring test files.

File: Common/src/main/java/dk/sdu/sesem4/common/data/EntityParts/MovingPart.java [hunk 4]
```diff
@@ -10,17 +10,19 @@
  * Moving part, that handles and represents moving in the game
  */
 public class MovingPart implements EntityPart {
-
-    /**
-     * The Knockback of the Entity
-     */
```
[the remark this answers is not in the document]
Comment (jakob753951) [reply Common/src/main/java/dk/sdu/sesem4/common/data/EntityParts/MovingPart.java:16]: It is self-describing

File: Collision/src/main/java/dk/sdu/sesem4/collision/CollisionSystem.java [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+import dk.sdu.sesem4.common.data.EntityParts.PositionPart;
+import dk.sdu.sesem4.common.data.entity.Entity;
+import dk.sdu.sesem4.common.data.gamedata.GameData;
+import dk.sdu.sesem4.common.data.math.Rectangle;
+import dk.sdu.sesem4.common.data.process.Priority;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class CollisionSystem implements PostProcessingServiceSPI {
```
[the remark this answers is not in the document]
Comment (jakob753951) [reply Collision/src/main/java/dk/sdu/sesem4/collision/CollisionSystem.java:16]: Has now been added

File: Collision/src/main/java/dk/sdu/sesem4/collision/Collision.java [hunk 6]
```diff
@@ -1,4 +1,13 @@
 package dk.sdu.sesem4.collision;
 
+import dk.sdu.sesem4.common.data.entity.Entity;
+
 public class Collision {
```
[the remark this answers is not in the document]
Comment (jakob753951) [reply Collision/src/main/java/dk/sdu/sesem4/collision/Collision.java:10]: Has been added

File: Map/src/main/java/dk/sdu/sesem4/map/MapProcessingService.java (hunk 2 above)
Comment (jakob753951) [reply Map/src/main/java/dk/sdu/sesem4/map/MapProcessingService.java:143]: Has been added

**Review by Muhammed-als — APPROVED** [review]

**Review by The0mikkel — APPROVED** [review]

**Review by JfriisJ — APPROVED** [review]

**Pull request merged**